superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -218,8 +318,12 @@ const DatabaseConnectionForm = ({
             key === 'database_name',
         ).map(field =>
           FORM_FIELD_MAP[field]({
-            required: parameters.required.includes(field),
-            changeMethods: { onParametersChange, onChange },
+            required: parameters.required?.includes(field),
```

Comment:
I think this will cause some refactoring, especially thinking about SSL (I don't think encryption is required)

Reply:
this only sets the required field for the `FORM_FIELD_MAP`

https://github.com/apache/superset/pull/14829/files#diff-ec023fde364c517087ecce2de11eb7374679b99a8afc2c641cab1df94eb9a2fcR60